### PR TITLE
EventSystem: constexpr conversions

### PR DESCRIPTION
**Pull request by neheb**
Small size reduction.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: main/EventSystem.h [hunk 1]
```diff
@@ -201,8 +201,9 @@ class CEventSystem : public CLuaCommon, StoppableTask, CNotificationObserver
 	std::string m_lua_Dir;
 	std::string m_szStartTime;
 
-	static const std::string m_szReason[], m_szSecStatus[];
-	static const _tJsonMap JsonMap[];
+	static const std::array<const char *, 7> m_szReason;
```
Comment (gizmocuz) [inline main/EventSystem.h:204]: If we rename m_szSecStatus to m_szEventsystemSecStatus , we could remove them from the header file.
(Same for the other array's)
The problem now/currently is that you need to specify the size on two locations (header and source file)
Declaring them only in the source file has also the benefit that no header needs to be changed and less source files need to be recompiled after a change
Is it possible to do this in the PR as well ?

Comment (neheb) [reply main/EventSystem.h:204]: Doesn't look like it. It's being used in main/dzVents.cpp

**Comment by gizmocuz** [comment]
Yes, but you can declare it as

extern m_szEventsystemSecStatus[]

in dzVents ?

**Comment by neheb** [comment]
I'll try that.

**Comment by gizmocuz** [comment]
Do we need to declare the size ? If yes, then we should leave it like it is, otherwise it would be a plus ;)

**Comment by neheb** [comment]
For std::array , yes. C++17 has CTAD which avoids the need for it.

**Comment by gizmocuz** [comment]
Then I am afraid you need to declare the size in main/dzVents.cpp as well.
Then we better leave it like it is now, thanks ! ;)

**Comment by neheb** [comment]
Yeah.

```
main/dzVents.cpp:758:60: error: no member named 'm_szSecStatus' in 'CEventSystem'; did you mean 'm_SecStatus'?
        luaTable.AddString("Security", m_mainworker.m_eventsystem.m_szSecStatus[secStatus]);
```

Has to be a member of CEventSystem

**Comment by gizmocuz** [comment]
Thank you once again !!

**Pull request merged**